classes/submission/DAO.inc.php

```diff
@@ -163,15 +163,38 @@ public function getIdsBySetting(string $settingName, $settingValue, int $context
      */
     public function getByPubId(string $pubIdType, string $pubId, $contextId = null): ?Submission
     {
-        $qb = DB::table('publication_settings ps')
-            ->join('publications p', 'p.publication_id', '=', 'ps.publication_id')
-            ->join('submissions s', 'p.publication_id', '=', 's.current_publication_id')
-            ->where('ps.setting_name', '=', 'pub-id::' . $pubIdType)
-            ->where('ps.setting_value', '=', $pubId);
-
-        if ($contextId) {
-            $qb->where('s.context_id', '=', (int) $contextId);
+        // Add check for incoming DOI request for legacy calls that bypass the Submission Repository
+        if ($pubIdType == 'doi') {
+            return Repo::submission()->getByPubId('doi', $pubId, $contextId);
+        } else {
+            $qb = DB::table('publication_settings ps')
+                ->join('publications p', 'p.publication_id', '=', 'ps.publication_id')
+                ->join('submissions s', 'p.publication_id', '=', 's.current_publication_id')
+                ->where('ps.setting_name', '=', 'pub-id::' . $pubIdType)
+                ->where('ps.setting_value', '=', $pubId);
+
+            if ($contextId) {
+                $qb->where('s.context_id', '=', (int) $contextId);
+            }
+
+            $row = $qb->get(['s.submission_id']);
+
+            return $row
+                ? $this->get($row->submission_id)
+                : null;
         }
+    }
+
+    /**
+     * Retrieve a submission by the current publication's doiId
+     *
+     *
+     */
+    public function getByDoiId(int $doiId): ?Submission
+    {
+        $qb = DB::table('publications', 'p')
+            ->leftJoin('submissions as s', 's.current_publication_id', '=', 'p.publication_id')
+            ->where('p.doi_id', '=', $doiId);
```

Comment:
It looks to me like these two methods exist across all the objects, but `getByDoiId` is only used by `getByPubId` and `getByPubId` is only used by the `Resolver` plugin for DOIs at the article level. Maybe we can simplify this to just a `getByDoi(string $doi, Context $context)` method that the `Resolver` plugin calls?